### ISSUE TITLE
Debug Gradient Flow

### ISSUE THREAD

**Issue by MartinLNG** [MartinLNG/adversarial_training_mps#107]
Seems like gradients do not flow to MPS during GAN training. Variation in GAN loss might be due only to changes in discriminator. Variation in classification loss on validation set might be due reininitalizations of MPS and machine accuracy.
